diagnostics-app/src/app/events/events-card/events-card.component.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Defines the Card Component in Events.
+ * Imported by events.module.ts
+ */
+
+import {
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnInit
+} from '@angular/core';
+
+import {
+  EventsCardState,
+  EventCategory,
+} from '@common/dpsl';
+import { ResponseErrorInfoMessage } from '@common/message';
+import { EventsService } from 'src/app/core/services/events.service';
+
+@Component({
+  selector: 'app-events-card',
+  templateUrl: './events-card.component.html',
+  styleUrls: ['./events-card.component.css'],
+})
+export class EventsCardComponent implements OnInit {
+  @Input({ required: true }) category!: EventCategory;
+
+  public eventList: object[] = [];
+  public error?: String; // the error message received, null if no error occurs
+  public state? = EventsCardState.NOT_LISTENING;
+
+  constructor(private eventsService: EventsService, private changeDetectorRef: ChangeDetectorRef) {}
+
+  ngOnInit(): void {
+    this.eventsService.getSubject(this.category).then(
+      res => {
+        if (res.success && res.subject) {
+          res.subject.subscribe({
+            next: event => {
+              this.error = undefined;
+              this.eventList.push(event);
+
+              // this is for triggering Angular change detection
+              this.changeDetectorRef.detectChanges();
+            }
+          })
+        } else {
+          this.error = res.error;
+        }
+      }
+    ).catch(err => {
+      this.error = err.message;
+    });
+  }
+
+  startCapturingEvents = async () => {
```

Comment:
Add a comment to say that `state` must be `NOT_LISTENING` when `startCapturingEvents` is called. Ditto to `stopCapturingEvents`.

Reply:
From the UI side we disable the button once we start listening, so we can't issue the same call twice. However, if somehow the function is called twice, then we will see the event output being cleared.

Let me add a check to only call start or stop when the state is NOT_LISTENING or LISTENING instead.

Reply:
Thanks. The check looks good to me.